inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -577,11 +577,30 @@ void MKLDNNGraphOptimizer::FuseConvolutionAndZeroPoints(MKLDNNGraph &graph) {
     }
 }
 
-static bool BF16QuantizeNodeFusing(MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
-    return childNode->getType() == FakeQuantize &&
+static void BF16QuantizeNodeFusing(MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
+    // TODO: We have to extend gemm_bf16_convolution_fwd_t, jit_avx512_core_bf16_1x1_conv_kernel,
+    // _jit_avx512_core_bf16_fwd_kernel and jit_avx512_dw_conv_fwd_kernel_bf16 from oneDNN
+    // to support Quantization in post ops
+    auto bf16ToFP32 = [](MKLDNNNodePtr node) {
+        for (size_t i = 0; i < node->getOriginalInputsNumber(); i++) {
+            if (node->getOriginalInputPrecisionAtPort(i) == Precision::BF16) {
+                node->setOriginalInputPrecisionAtPort(i, Precision::FP32);
+            }
+        }
+
+        for (size_t i = 0; i < node->getOriginalOutputsNumber(); i++) {
+            if (node->getOriginalOutputPrecisionAtPort(i) == Precision::BF16) {
+                node->setOriginalOutputPrecisionAtPort(i, Precision::FP32);
+            }
+        }
+    };
+    if (childNode->getType() == FakeQuantize &&
         one_of(Precision::BF16,
             parentNode->getOriginalOutputPrecisionAtPort(0),
-            childNode->getOriginalOutputPrecisionAtPort(0));
+            childNode->getOriginalOutputPrecisionAtPort(0))) {
+        bf16ToFP32(parentNode);
```

Comment:
If I get the solution right you are forcing parentNode to be executed in FP32 precision in case FQ was fused. Why exactly this behavior was chosen? Why cannot we just disable parent and child fusion for this scenario?
 
cc'ed @maxnick